How to set a custom host in publishToMavenCentral?

```
fun publishToMavenCentral(host: SonatypeHost = SonatypeHost.DEFAULT, automaticRelease: Boolean = false) {
```

SonatypeHost is an enum, so how to set a custom host in the form of xxx.oss.sonatype.org for publishing to Sonetype?

DEFAULT and use SONATYPE_HOST env? Can the enum be opened up to a value type?

That doesn't work.

```
Caused by: java.lang.IllegalArgumentException: No enum constant com.vanniktech.maven.publish.SonatypeHost.google.oss.sonatype.org
        at com.vanniktech.maven.publish.SonatypeHost.valueOf(SonatypeHost.kt)
        at com.vanniktech.maven.publish.MavenPublishPlugin.apply(MavenPublishPlugin.kt:20)
        at com.vanniktech.maven.publish.MavenPublishPlugin.apply(MavenPublishPlugin.kt:11)
```

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt
@@ -81,7 +81,8 @@ abstract class MavenPublishBaseExtension(
   }
 
   @JvmOverloads
-  fun publishToMavenCentral(host: String = "DEFAULT", automaticRelease: Boolean = false) {
+  @JvmSynthetic
+  fun publishToMavenCentral(host: String, automaticRelease: Boolean = false) {
     publishToMavenCentral(SonatypeHost.valueOf(host), automaticRelease)
   }
 

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt
@@ -80,6 +80,11 @@ abstract class MavenPublishBaseExtension(
     project.tasks.registerDropRepository(buildService)
   }
 
+  @JvmOverloads
+  fun publishToMavenCentral(host: String = "DEFAULT", automaticRelease: Boolean = false) {
+    publishToMavenCentral(SonatypeHost.valueOf(host), automaticRelease)
+  }
+
   /**
    * Automatically apply Gradle's `signing` plugin and configure all publications to be signed. If signing credentials
    * are not configured this will fail the build unless the current version is a `SNAPSHOT`.

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
@@ -8,12 +8,9 @@ import org.gradle.api.provider.Provider
  *
  * https://central.sonatype.org/articles/2021/Feb/23/new-users-on-s01osssonatypeorg/
  */
-enum class SonatypeHost(
+class SonatypeHost(
   internal val rootUrl: String
 ) {
-  DEFAULT("https://oss.sonatype.org"),
-  S01("https://s01.oss.sonatype.org");
-
   internal fun apiBaseUrl(): String {
     return "$rootUrl/service/local/"
   }
@@ -27,5 +24,16 @@ enum class SonatypeHost(
     } else {
       "$rootUrl/service/local/staging/deployByRepositoryId/${stagingRepositoryId.get()}/"
     }
+  }
+
+  companion object {
+    fun valueOf(sonatypeHost: String): SonatypeHost = when (sonatypeHost) {
+      "DEFAULT" -> DEFAULT
+      "S01" -> S01
+      else -> throw IllegalArgumentException("No SonatypeHost constant $sonatypeHost")
+    }
+
+    val DEFAULT = SonatypeHost("https://oss.sonatype.org")
+    val S01 = SonatypeHost("https://s01.oss.sonatype.org")
   }
 }

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
@@ -1,5 +1,6 @@
 package com.vanniktech.maven.publish
 
+import java.io.Serializable
 import org.gradle.api.provider.Provider
 
 /**
@@ -10,7 +11,7 @@ import org.gradle.api.provider.Provider
  */
 data class SonatypeHost(
   internal val rootUrl: String
-) {
+): Serializable {
   internal fun apiBaseUrl(): String {
     return "$rootUrl/service/local/"
   }

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
@@ -8,7 +8,7 @@ import org.gradle.api.provider.Provider
  *
  * https://central.sonatype.org/articles/2021/Feb/23/new-users-on-s01osssonatypeorg/
  */
-class SonatypeHost(
+data class SonatypeHost(
   internal val rootUrl: String
 ) {
   internal fun apiBaseUrl(): String {

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
@@ -28,13 +28,17 @@ data class SonatypeHost(
   }
 
   companion object {
+    @JvmStatic
     fun valueOf(sonatypeHost: String): SonatypeHost = when (sonatypeHost) {
       "DEFAULT" -> DEFAULT
       "S01" -> S01
       else -> throw IllegalArgumentException("No SonatypeHost constant $sonatypeHost")
     }
 
+    @JvmField
     val DEFAULT = SonatypeHost("https://oss.sonatype.org")
+
+    @JvmField
     val S01 = SonatypeHost("https://s01.oss.sonatype.org")
   }
 }

--- a/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
+++ b/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
@@ -11,7 +11,7 @@ import org.gradle.api.provider.Provider
  */
 data class SonatypeHost(
   internal val rootUrl: String
-): Serializable {
+) : Serializable {
   internal fun apiBaseUrl(): String {
     return "$rootUrl/service/local/"
   }

